core/src/main/java/com/linecorp/armeria/common/logging/DefaultRequestLog.java

```diff
@@ -438,29 +440,39 @@ private static void completeSatisfiedFutures(RequestLogFuture[] satisfiedFutures
 
     @Nullable
     private RequestLogFuture[] removeSatisfiedFutures() {
-        if (pendingFutures.isEmpty()) {
-            return null;
+        reentrantLock.lock();
```

Comment:
This method is actually called from inside a lock, and doesn't have to be guarded.

What do you think of making this method `static` instead?

e.g.
```
+    private RequestLogFuture[] removeSatisfiedFutures(List<RequestLogFuture> pendingFutures) {
+        if (pendingFutures.isEmpty()) {
+            return null;
         }
 
         final int flags = this.flags;
-        reentrantLock.lock();
-        try {
-            final int maxNumListeners = pendingFutures.size();
-            final Iterator<RequestLogFuture> i = pendingFutures.iterator();
-            RequestLogFuture[] satisfied = null;
-            int numSatisfied = 0;
-
-            do {
-                final RequestLogFuture e = i.next();
-                final int interestedFlags = e.interestedFlags;
-                if ((flags & interestedFlags) == interestedFlags) {
-                    i.remove();
-                    if (satisfied == null) {
-                        satisfied = new RequestLogFuture[maxNumListeners];
-                    }
-                    satisfied[numSatisfied++] = e;
+        final int maxNumListeners = pendingFutures.size();
+        final Iterator<RequestLogFuture> i = pendingFutures.iterator();
+        RequestLogFuture[] satisfied = null;
+        int numSatisfied = 0;
+
+        do {
+            final RequestLogFuture e = i.next();
+            final int interestedFlags = e.interestedFlags;
+            if ((flags & interestedFlags) == interestedFlags) {
+                i.remove();
+                if (satisfied == null) {
+                    satisfied = new RequestLogFuture[maxNumListeners];
                 }
-            } while (i.hasNext());
+                satisfied[numSatisfied++] = e;
+            }
+        } while (i.hasNext());
 
-            return satisfied;
-        } finally {
-            reentrantLock.unlock();
-        }
+        return satisfied;
     }
```

Reply:
Fixed as you mentioned, sorry for late😅